app/shared/types/templateSchema.ts

```diff
@@ -155,12 +155,70 @@ ajv.addKeyword('cantDeleteInheritedProperties', {
   },
 });
 
+ajv.addKeyword('cantReuseNameWithDifferentType', {
+  async: true,
+  errors: true,
+  type: 'object',
+  async validate(_schema: any, template: TemplateSchema) {
+    if (!template.properties || template.properties.length === 0) {
+      return true;
+    }
+
+    const condition = template.properties.map(property => ({
+      $and: [
+        {
+          name: property.name,
+          $or: [
+            { content: { $ne: property.content } },
+            { type: { $ne: property.type } },
+            { relationtype: { $ne: property.relationType } },
+          ],
+        },
+      ],
+    }));
+    const query = { properties: { $elemMatch: { $or: [...condition] } } };
+    const matchedTemplates = await model.get(query);
+    const otherTemplates = matchedTemplates.filter(t => !t._id.equals(template._id));
+    if (otherTemplates.length > 0) {
+      const errorProperties = template.properties.reduce((propertyNames: string[], property) => {
+        otherTemplates.forEach((t: TemplateSchema) => {
+          const matches = t.properties?.find(
+            p =>
+              p.name === property.name &&
+              (p.content !== property.content ||
+                p.type !== property.type ||
+                p.relationtype !== property.relationType)
+          );
+          if (matches && !propertyNames.includes(property?.name || '')) {
+            propertyNames.push(property?.name || '');
+          }
+        });
+        return propertyNames;
+      }, []);
+
+      throw new Ajv.ValidationError(
+        errorProperties.map(property => ({
+          keyword: 'cantReuseNameWithDifferentType',
+          schemaPath: '',
+          params: { keyword: 'cantReuseNameWithDifferentType' },
+          message:
+            'Entered label is already in use on another property with a different type or thesaurus',
+          dataPath: `.properties.${property}`,
```

Comment:
@mfacar,  Ajv error can also have data property, i suggest you include the data that is wrong, this way there will be better information on why the error happens. you can include the properties that do not pass the validation from the current template that is being validated.

Reply:
@daneryl can you elaborate on this please? We are already sending the failing property in the dataPath (which is also shown in the client) and we are generating one error message per failed field.

Reply:
@fnocetti including the data that actually fails validation will provide better feeback, the actual property, maybe even the properties that have the same name and why this one is not valid, but i guess that is easily debuggable with the current error, this should be enough as it is. LGTM